x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/form.tsx

```diff
@@ -291,6 +311,41 @@ export function useOnSubmit({
     }
   }, [packagePolicy, agentPolicies, updatePackagePolicy, canUseMultipleAgentPolicies]);
 
+  const { handleSetupTechnologyChange, selectedSetupTechnology } = useSetupTechnology({
```

Comment:
I moved the `useSetupTechnology` inside the main hook, it was outside and was causing unexpected behaviours with state updates.